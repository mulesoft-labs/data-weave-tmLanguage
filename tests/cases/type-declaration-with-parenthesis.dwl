type Test = (String | Boolean)
type Test2 = String
type Test4 = (String | Boolean | (String, Boolean) -> Boolean)
type Test5 = (String | Boolean | (String, Boolean) -> (Boolean))
type Test6 = ((String, Boolean) -> (Boolean))
type Test7 = (String | Boolean | (String | Boolean) | Boolean) -> Boolean