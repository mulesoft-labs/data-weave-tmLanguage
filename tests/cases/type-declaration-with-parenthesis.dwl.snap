>type Test = (String | Boolean)
#^^^^ source.data-weave meta.directive.type.dw storage.type.dw
#    ^ source.data-weave meta.directive.type.dw
#     ^^^^ source.data-weave meta.directive.type.dw entity.name.type.dw
#         ^ source.data-weave meta.directive.type.dw
#          ^ source.data-weave meta.directive.type.dw keyword.other.dw
#           ^ source.data-weave meta.directive.type.dw
#            ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#             ^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                   ^ source.data-weave meta.directive.type.dw
#                    ^ source.data-weave meta.directive.type.dw keyword.operator.declaration.dw
#                     ^ source.data-weave meta.directive.type.dw
#                      ^^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                             ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
>type Test2 = String
#^^^^ source.data-weave meta.directive.type.dw storage.type.dw
#    ^ source.data-weave meta.directive.type.dw
#     ^^^^^ source.data-weave meta.directive.type.dw entity.name.type.dw
#          ^ source.data-weave meta.directive.type.dw
#           ^ source.data-weave meta.directive.type.dw keyword.other.dw
#            ^ source.data-weave meta.directive.type.dw
#             ^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
>type Test4 = (String | Boolean | (String, Boolean) -> Boolean)
#^^^^ source.data-weave meta.directive.type.dw storage.type.dw
#    ^ source.data-weave meta.directive.type.dw
#     ^^^^^ source.data-weave meta.directive.type.dw entity.name.type.dw
#          ^ source.data-weave meta.directive.type.dw
#           ^ source.data-weave meta.directive.type.dw keyword.other.dw
#            ^ source.data-weave meta.directive.type.dw
#             ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#              ^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                    ^ source.data-weave meta.directive.type.dw
#                     ^ source.data-weave meta.directive.type.dw keyword.operator.declaration.dw
#                      ^ source.data-weave meta.directive.type.dw
#                       ^^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                              ^ source.data-weave meta.directive.type.dw
#                               ^ source.data-weave meta.directive.type.dw keyword.operator.declaration.dw
#                                ^ source.data-weave meta.directive.type.dw
#                                 ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#                                  ^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                                        ^^ source.data-weave meta.directive.type.dw
#                                          ^^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                                                 ^^^^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#                                                     ^ source.data-weave meta.directive.type.dw
#                                                      ^^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                                                             ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
>type Test5 = (String | Boolean | (String, Boolean) -> (Boolean))
#^^^^ source.data-weave meta.directive.type.dw storage.type.dw
#    ^ source.data-weave meta.directive.type.dw
#     ^^^^^ source.data-weave meta.directive.type.dw entity.name.type.dw
#          ^ source.data-weave meta.directive.type.dw
#           ^ source.data-weave meta.directive.type.dw keyword.other.dw
#            ^ source.data-weave meta.directive.type.dw
#             ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#              ^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                    ^ source.data-weave meta.directive.type.dw
#                     ^ source.data-weave meta.directive.type.dw keyword.operator.declaration.dw
#                      ^ source.data-weave meta.directive.type.dw
#                       ^^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                              ^ source.data-weave meta.directive.type.dw
#                               ^ source.data-weave meta.directive.type.dw keyword.operator.declaration.dw
#                                ^ source.data-weave meta.directive.type.dw
#                                 ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#                                  ^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                                        ^^ source.data-weave meta.directive.type.dw
#                                          ^^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                                                 ^^^^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#                                                     ^ source.data-weave meta.directive.type.dw
#                                                      ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#                                                       ^^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                                                              ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#                                                               ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
>type Test6 = ((String, Boolean) -> (Boolean))
#^^^^ source.data-weave meta.directive.type.dw storage.type.dw
#    ^ source.data-weave meta.directive.type.dw
#     ^^^^^ source.data-weave meta.directive.type.dw entity.name.type.dw
#          ^ source.data-weave meta.directive.type.dw
#           ^ source.data-weave meta.directive.type.dw keyword.other.dw
#            ^ source.data-weave meta.directive.type.dw
#             ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#              ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#               ^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                     ^^ source.data-weave meta.directive.type.dw
#                       ^^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                              ^^^^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#                                  ^ source.data-weave meta.directive.type.dw
#                                   ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#                                    ^^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                                           ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#                                            ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
>type Test7 = (String | Boolean | (String | Boolean) | Boolean) -> Boolean
#^^^^ source.data-weave meta.directive.type.dw storage.type.dw
#    ^ source.data-weave meta.directive.type.dw
#     ^^^^^ source.data-weave meta.directive.type.dw entity.name.type.dw
#          ^ source.data-weave meta.directive.type.dw
#           ^ source.data-weave meta.directive.type.dw keyword.other.dw
#            ^ source.data-weave meta.directive.type.dw
#             ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#              ^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                    ^ source.data-weave meta.directive.type.dw
#                     ^ source.data-weave meta.directive.type.dw keyword.operator.declaration.dw
#                      ^ source.data-weave meta.directive.type.dw
#                       ^^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                              ^ source.data-weave meta.directive.type.dw
#                               ^ source.data-weave meta.directive.type.dw keyword.operator.declaration.dw
#                                ^ source.data-weave meta.directive.type.dw
#                                 ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#                                  ^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                                        ^ source.data-weave meta.directive.type.dw
#                                         ^ source.data-weave meta.directive.type.dw keyword.operator.declaration.dw
#                                          ^ source.data-weave meta.directive.type.dw
#                                           ^^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                                                  ^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#                                                   ^ source.data-weave meta.directive.type.dw
#                                                    ^ source.data-weave meta.directive.type.dw keyword.operator.declaration.dw
#                                                     ^ source.data-weave meta.directive.type.dw
#                                                      ^^^^^^^ source.data-weave meta.directive.type.dw support.class.dw
#                                                             ^^^^ source.data-weave meta.directive.type.dw keyword.operator.grouping.dw
#                                                                 ^ source.data-weave meta.directive.type.dw
#                                                                  ^^^^^^^ source.data-weave meta.directive.type.dw support.class.dw